Creating Pull Request from my branch to development branch

BorisMurashkevich opened the pull request.

BorisMurashkevich committed:
Adding code to the branch

wojtekand commented on index.html:
```diff
@@ -0,0 +1,106 @@
+    }
+    .YInput{
+        width: 30px;
+    }
```
create separate file for css

wojtekand commented on arraysForDeletion.js:
```diff
@@ -0,0 +1,246 @@
+function arraysForDeletion(){
```
use classes ES6 with classess instead of functions. Add description to each function starting from multiline comment.

wojtekand commented on arraysForDeletion.js:
```diff
@@ -0,0 +1,246 @@
+function arraysForDeletion(){
+    let emptyTab = []
+    let emptyTabCopy = []
```
create class properties

wojtekand commented on arraysForDeletion.js:
```diff
@@ -0,0 +1,246 @@
+        if (emptyEl.innerHTML === ' '){                         // making an array with empty fields
+            emptyTab.push(board1.tableId[i])
+        }
+    }
```
create function from above code

wojtekand commented on arraysForDeletion.js:
```diff
@@ -0,0 +1,246 @@
+        strChange = strChange.replace("td","")
+        emptyTabCopy[i] = strChange
+        emptyTabCopy[i] = Number(emptyTabCopy[i])
+    }
```
same here

wojtekand commented on arraysForDeletion.js:
```diff
@@ -0,0 +1,246 @@
+    let leftArraysDone = []
+    let rightArraysDone = []
+    let leftEdgesInt = []
+    let rightEdgesInt = []
```
declare class properties

wojtekand commented on arraysForDeletion.js:
```diff
@@ -0,0 +1,246 @@
+            allBoard.push(i)
+        }
+        let addNumToArray = 0
+        leftEdgesInt.push(addNumToArray) //left edge of the board always starts with 0
```
use airbnb linter for code rules and formatting https://www.npmjs.com/package/eslint-config-airbnb

wojtekand commented on arraysForDeletion.js:
```diff
@@ -0,0 +1,246 @@
+    for (let i=0;i<rightEdgesInt.length;i++){
+        let numberDelete2 = tabsIntWithoutEdges.indexOf(rightEdgesInt[i])
+        tabsIntWithoutEdges.splice(numberDelete2,1)
+    }
```
use objective programming. Create class which will be representing the game. Read about objective programming first to get the concept of it.

wojtekand commented on arraysForDeletion.js:
```diff
@@ -0,0 +1,246 @@
+
+                }
+            }
+        }
```
create separate functions from above code. Each functions should be atomic do one thing not multiple things. For example `initializeMinesBoard` which will put randomly mines on the field.

wojtekand commented on arraysForDeletion.js:
```diff
@@ -0,0 +1,246 @@
+        arrayLeft = arrayLeft.filter((item, index) => {
+            return arrayLeft.indexOf(item) === index
+        })
+        function tabsSeparation(tempArraySideSeparate,edges,array) {
```
name function base on what they are doing

wojtekand commented on arraysForDeletion.js:
```diff
@@ -0,0 +1,246 @@
+
+
+
+
```
remove empty lines. Your code has to look good as well and be readable. You are not writing code to yourself.

wojtekand commented on arraysForDeletion.js:
```diff
@@ -0,0 +1,246 @@
+        tempArrayRightSeparate = Array.from(set).map(JSON.parse);
+
+        let arrayLeftFinal = []
+        function arraySep(inArray,outArray,side) {
```
name function base on what the function is doing for example `showFieldsWithNoMinesAround` which will show all the fields around the fields which contains zero mines around. BTW you should have this function made using recursion in your logic.

BorisMurashkevich committed:
Changes from pull request commits were made

wojtekand commented on .idea/misc.xml:
```diff
@@ -1,6 +1,6 @@
-    <option name="languageLevel" value="ES6" />
+    <option name="languageLevel" value="FLOW" />
   </component>
 </project>
```
above files should be added to gitignore

wojtekand commented on app.js:
```diff
@@ -0,0 +1,462 @@
+valueLoad() // loading entered size of the board
+class board {
```
zainstaluj eslinta i airbnb rules do niego.

wojtekand replied:
class name starts with capital letter

wojtekand commented on app.js:
```diff
@@ -0,0 +1,462 @@
+minesAmount = Math.floor(minesAmount)
+document.getElementById("minesCounter").value = minesAmount
+let infoSize = 60*board1.x+60+board1.x*3.2+'px'     // formula for measuring size of the info menu (the one with the counter and stuff)
+document.getElementById("info").style.width = infoSize
```
move above functions to separate class. Put all classes in separate files.

wojtekand commented on app.js:
```diff
@@ -0,0 +1,462 @@
+        this.size = x*y
+
+    }
+    creatingTable(x,y){
```
zmien nazwe na `createBoard` nie dodawaj `ing` w nazwach. Metoda ma wydawac polecenie.

wojtekand commented on app.js:
```diff
@@ -0,0 +1,462 @@
+
+            }
+            this.table.appendChild(tr);
+        }
```
Board should contain list of fields. Board should have render method which will invoke render on each field. Please create class diagram this way you will understand connections between classes. Please remember about SOLID principle when you are designing class diagram and most important single responsibility of each class. Change the whole code according to this comment.

wojtekand commented on app.js:
```diff
@@ -0,0 +1,462 @@
+class field {
+    emptyTabCopy = []
+    randomMinesTemp = []
+    mines = []
```
Field may contain single mine not list of mines. Look at classes like on lego blocks. You build an application from them. If you take car for example then car is build from doors, wheels, engine and etc. all those should be a separate class. This way you design application. Below code has to be adjusted to that.

BorisMurashkevich committed:
New classes have been added

New classes, minor changes through the code. Class Field have been completely redisigned

UlanSoftware commented on ShowEmptyFields.js:
```diff
@@ -0,0 +1,372 @@
+    }
+  }
+
+  showFieldsWithNoMinesAround() {
```
this function is overcomplicated. Please use recursion this will make logic much more simplier.

UlanSoftware commented on ShowEmptyFields.js:
```diff
@@ -0,0 +1,372 @@
+    }
+  }
+
+  fixEmptyFields() {
```
this function as well is to complicated. You need to write simple code. Having 4 for loops one inside the other that's to much.

UlanSoftware commented on board.js:
```diff
@@ -0,0 +1,47 @@
+const load = new LoadAndSaveTheBoard();
+load.valueLoad(); // loading the previous selected size of the board
+class Board {
```
Remember each object should have some responsibility according to SOLID principle. This one represents the board of the game so all logic connected with board should be kept over here. Board should contains two dimensional array with Fields.

UlanSoftware commented on board.js:
```diff
@@ -0,0 +1,47 @@
+    this.size = x * y;
+  }
+
+  createBoard(x, y) {
```
thats much better

UlanSoftware commented on board.js:
```diff
@@ -0,0 +1,47 @@
+        buttonColor.style.background = 'grey';
+
+        const tdId = `${this.btnId}td`; // giving an id to each element of the row
+        td.id = tdId;
```
the whole logic connected with rendering field should be part of Field class and more precisely createCell method according to snigle principle rule.

UlanSoftware commented on board.js:
```diff
@@ -0,0 +1,47 @@
+const load = new LoadAndSaveTheBoard();
+load.valueLoad(); // loading the previous selected size of the board
+class Board {
+  tableId = []; // this array contains list of fields
```
this table should contain Fields not ids

UlanSoftware commented on field.js:
```diff
@@ -0,0 +1,21 @@
+  btn = document.createElement('button');
+
+  createCell() {
+    this.td.style.border = '1px solid black';
```
keep styles in css file

UlanSoftware commented on field.js:
```diff
@@ -0,0 +1,21 @@
+  }
+
+  createButton() {
+    document.body.appendChild(this.btn);
```
button should be appended to td element so this method should be invoke by createCell

UlanSoftware commented on field.js:
```diff
@@ -0,0 +1,21 @@
+class Field {
+  td = document.createElement('td');
+
```
field should contain state property if hidden or shown, it should also contain information if it contains the bomb or how many bombs are arround it. Base on those information it should be rendered differentyl.

UlanSoftware commented on field.js:
```diff
@@ -0,0 +1,21 @@
+    this.btn.style.width = '60px';
+    this.btn.style.height = '60px';
+    this.btn.style.marginLeft = '-4.5em';
+    return this.btn;
```
button should have listener attached

UlanSoftware commented on game.js:
```diff
@@ -0,0 +1,217 @@
+          }
+        }
+      }
+    });
```
this code once again is overcomplicated. Most of the problems are connected with wrong data structures. Please redesign it according to mu suggestion. Those kind of code is not acceptable please keep it simple.

BorisMurashkevich committed:
Code has been siplified and redisigned using new gata structures

wojtekand commented on board.js:
```diff
@@ -0,0 +1,97 @@
+
+  x;
+
+  y;
```
as I understand this is sizeX and sizeY. Please create meaningful names

wojtekand commented on board.js:
```diff
@@ -0,0 +1,97 @@
+
+  fieldsList = [];
+
+  emptyFieldsList = []; // two-dimensional array with empty fields
```
you dont need to have two list. Field should contain information if it is empty or not.

wojtekand commented on board.js:
```diff
@@ -0,0 +1,97 @@
+
+  allEdges = [];
+
+  corners = [];
```
you dont require above  arrays. I mean leftEdges, rightEdges, corners, ... . By using sizeX and sizeY you know if field is on left or right or top or bottom edge. You know that by comparing x == sizeX if it is true you know it is right edget if x == 0 it is left edge. By creating those tables you only make logic much more complikcated.

wojtekand commented on board.js:
```diff
@@ -0,0 +1,97 @@
+        this.fieldsList.push(field);
+        const td = field.createCell();
+        tr.appendChild(td);
+        const btn = field.createButton();
```
field.createButton() should be inside field.createCell(). You have to remember about single responsibility principal. Field is responsible for creating it. So logic which creates field should be only in Field clase you shouldnt split it between different classes.

wojtekand commented on board.js:
```diff
@@ -0,0 +1,97 @@
+    }
+  }
+
+  getEdgesOfTheBoard() {
```
you dont require this funtion

wojtekand commented on field.js:
```diff
@@ -0,0 +1,172 @@
+
+  isLeftEdge = false;
+
+  isRightEdge = false;
```
you dont need isLeftEdge and isRightEdge you can calculate it.

wojtekand commented on field.js:
```diff
@@ -0,0 +1,172 @@
+  createButton() {
+    this.td.appendChild(this.btn);
+    this.btn.className = 'jsButtonsStyle';
+    this.btn.addEventListener('click', () => {
```
below logic is overcomplicated. You cant write this kind of code. What your code should do on click is:
- check if field contains a bomb if it has a bomb player has lost otherwise
-  check how many bombs is around
-  if there are zero bombs around it should invoke the same function on all members using recursion

wojtekand commented on game.js:
```diff
@@ -0,0 +1,76 @@
+  emptyTabCopy = [];
+
+  constructor() {
+    this.infoMenu();
```
the name of the function should contains always some activity. Please change the name to renderInfoMenu or something like that

wojtekand commented on game.js:
```diff
@@ -0,0 +1,76 @@
+
+  constructor() {
+    this.infoMenu();
+    this.getEmptyFields();
```
you dont need to have this function

wojtekand commented on game.js:
```diff
@@ -0,0 +1,76 @@
+  constructor() {
+    this.infoMenu();
+    this.getEmptyFields();
+    this.showFieldsWIthNoMinesAround();
```
over here you should invoke function renderBoard or something like that. We dont want to show fields with no mines around. First moves user has to guess. Beside if you would like to add something like that then it should be board responsibility.

wojtekand commented on loadAndSave.js:
```diff
@@ -0,0 +1,37 @@
+class LoadAndSaveTheBoard {
```
This class is fine

wojtekand commented on minePlace.js:
```diff
@@ -0,0 +1,114 @@
+class MinePlace {
```
Add comments to each class to show what is this class responsibility do the same for each property.

wojtekand commented on minePlace.js:
```diff
@@ -0,0 +1,114 @@
+    this.minePlace();
+  }
+
+  minePlace() {
```
this function is to big. Function should do single thing.

wojtekand commented on minePlace.js:
```diff
@@ -0,0 +1,114 @@
+      // placing mines and changing the field's property "mine"
+      board.fieldsList[this.mines[i]].mine = true;
+    }
+    for (let i = 0; i < board.size; i++) {
```
the number of the mines should be selected base on the difficulty level. And you should place selected number of mines on the board. So in that case there is no need to count mines.

wojtekand commented on minePlace.js:
```diff
@@ -0,0 +1,114 @@
+      }
+      if (minesCounter === '0') {
+        drawCounter.innerHTML = ' ';
+      }
```
above code is connected with field rendering. It should not be place over here.

BorisMurashkevich committed:
Quality of code improvements

Parts of code have been moved or changed accordingly to their responsibilities. Difficulty level was added

wojtekand commented on board.js:
```diff
@@ -1,35 +1,27 @@
 load.valueLoad(); // loading the previous selected size of the board
-class Board {
+class Board { // this class is responsible for creating the board and showing empty fields
```
use star comments over the class

wojtekand commented on board.js:
```diff
@@ -40,58 +32,170 @@ class Board {
+      indexOfCurrentElement++;
+      this.showFieldsWithNoMinesAround(indexOfCurrentElement, deleteButtons);
+    }
+  }
```
above logic is overcomplicated you can do it in much simpler way using reursive function.

wojtekand commented on board.js:
```diff
@@ -40,58 +32,170 @@ class Board {
+    }
+  }
+
+  separateEmptyFields(iteration) { // list with empty fields should be divided into
```
you dont need to create empty fields list. If you have the list with fields each field should contain information about its state and how many mines is around it. That should be base on your logic

wojtekand replied:
Beside once again I asked you to keep the logic simple. Dont have to long function.

wojtekand commented on board.js:
```diff
@@ -40,58 +32,170 @@ class Board {
-    for (let i = 0; i < this.fieldsList.length; i += this.x) {
-      // changing the property of field if this field is on the edge of the board
-      this.fieldsList[i].isLeftEdge = true;
+  showFieldsWithNoMinesAround(indexOfCurrentElement, deleteButtons) {
```
add comments to every function.

wojtekand commented on field.js:
```diff
@@ -11,162 +12,118 @@ class Field {
-            } else if (idOfButtonsToDelete[i] === board.lowerEdges[j]) {
-              board.fieldsList[currentEl + numbersAroundButtonDown[k]].stateProperty = 'shown';
+        for (let i = 0; i < board.emptyFieldsListSeparated.length; i++) {
+          for (let j = 0; j < board.emptyFieldsListSeparated[i].length; j++) {
```
why are you accessing board in button. Button doesnt need to know anything about the board. Please read about domain driven design and object oriented programming. Please create a class diagram and put it in your project. I think that will help you to understand how you should divide the responsibility of each class. You need to understand the concept of object oriented programing. Have a look to SOLID principle. Concentrate on understanding those concepts first and after that refactor your code. Or the best idea start from scratch and think your project can be created in simpler way using above principles. Understan recursive functions.

BorisMurashkevich committed:
Code has been refactored

Code has been refactored according to responsibilities of each class

BorisMurashkevich committed:
Changes from previous comments were made

Comments were added, mines reveal was refactored using recursion and a couple other small changes
